### ISSUE TITLE
documentation: clarify rules for using DEPENDS

### ISSUE THREAD

**Issue by wjwwood** [ros/catkin#491]
See:

https://github.com/ros-visualization/rviz/pull/661

**Comment by jack-oquin** [comment]
I just noticed that this issue is assigned to me. I am happy to work on it, but could use some guidance figuring out what should be changed and where.

@dirk-thomas:
@dgossow:

**Comment by wjwwood** [comment]
I don't remember assigning this to you, but then again I don't remember making the ticket either...

**Comment by jack-oquin** [comment]
That happens to me all the time. But, I'm a bit older than you are. :smile:

**Comment by jbohren** [comment]
Shouldn't `DEPENDS` carry LD_FLAGS and CFLAGS forward along with INCLUDE_DIRS and LIBS? Currently it does not: https://github.com/ros/catkin/blob/groovy-devel/cmake/catkin_package.cmake#L140

**Comment by wjwwood** [comment]
tl;dr The `DEPEND` tag is just syntactic sugar, we need to support `LD_FLAGS` and `CFLAGS` first.

Currently catkin doesn't have those as export options, before we export the `LD_FLAGS` and/or `CFLAGS` for dependencies listed in `DEPEND` section, we would first need to provide a generic mechanism for users to set their `LD_FLAGS` and `CFLAGS` for their package in the `catkin_package` call.

I opened a more generic issue here: https://github.com/ros/catkin/issues/541

**Comment by dirk-thomas** [comment]
Please clarify which documentation this ticket refers to and what kind of information you think it should contain.

**Comment by wjwwood** [comment]
Quoting @dirk-thomas:

> But you should fill a bug against catkin taht this doc should clearly state that it only works for stuff which sets the correct variable names: http://ros.org/doc/groovy/api/catkin/html/howto/system_library_dependencies.html#exporting-interfaces

So I guess it just needs a mention that `DEPENDS` only passes certain variables and only if they are named correctly.

**Comment by dirk-thomas** [comment]
This should be clarified in 78fe98adcd4df827fb2c5581f80a5eebb6ce7946. If not please comment with further suggestions.

**Comment by jack-oquin** [comment]
Looks OK to me. 

I would have said more in the first place if I'd know what to say.
